src/components/imageOptionsEditor/imageOptionsEditor.js

```diff
@@ -78,35 +89,29 @@ define(['globalize', 'dom', 'dialogHelper', 'emby-checkbox', 'emby-select', 'emb
         });
     }
 
-    function editor() {
-        this.show = function (itemType, options, availableOptions) {
-            return new Promise(function (resolve, reject) {
-                var xhr = new XMLHttpRequest();
-                xhr.open('GET', 'components/imageOptionsEditor/imageOptionsEditor.template.html', true);
-
-                xhr.onload = function (e) {
-                    var template = this.response;
-                    var dlg = dialogHelper.createDialog({
-                        size: 'small',
-                        removeOnClose: true,
-                        scrollY: false
-                    });
-                    dlg.classList.add('formDialog');
-                    dlg.innerHTML = globalize.translateDocument(template);
-                    dlg.addEventListener('close', function () {
-                        saveValues(dlg, options);
-                    });
-                    loadValues(dlg, itemType, options, availableOptions);
-                    dialogHelper.open(dlg).then(resolve, resolve);
-                    dlg.querySelector('.btnCancel').addEventListener('click', function () {
-                        dialogHelper.close(dlg);
-                    });
-                };
-
-                xhr.send();
+export class showEditor {
```

Comment:
I'm not really a fan of this.

The following structure would lead to less refactoring down the line (pseudo-code for conciseness, it likely requires other changes):
```javascript
class ImageOptionsEditor {
    #dialog; // Private property containing the built dialog
    constructor(itemType, options, availableOptions) {
        // Create the dialog and assigns it to this.#dialog
    }
    show() {
        dialogHelper.open(this.#dialog).then(resolve, resolve);
        this.#dialog.querySelector('.btnCancel').addEventListener('click', () => {
            dialogHelper.close(this.#dialog);
        }
    }
}